### PR TITLE
Dev: ocf.py: Flesh out python library for resource agents

### PR DESCRIPTION
**Pull request by krig**
This PR adds various helper functions (like `is_probe` and `ocf_exit_reason`) as well as a metadata generator and run loop.

It also adds a document describing how to use the library.

Usage example:

``` python
import ocf

def start_action(argument):
    print("The start action receives the argument as a parameter: {}".format(argument))


def main():
    metadata = ocf.Metadata("example-agent",
                            shortdesc="This is an example agent",
                            longdesc="An example of how to " +
                            "write an agent in Python using the ocf " +
                            "Python library.")
    metadata.parameter("argument",
                       shortdesc="Example argument",
                       longdesc="This argument is just an example.",
                       content_type="string",
                       default="foobar")
    metadata.action("start", timeout=60)
    ocf.run(metadata,
            handlers={
                "start": start_action
            })

if __name__ == "__main__":
    main()
```

### CODE REVIEW

**Comment by oalbrigt** [comment]
No need to rewrite the indentation, and it just makes it harder to track the changes.

**Comment by krig** [comment]
Leaving the indentation as tabs confuses my editor, and it's not according to PEP 8 :(
But I can revert that part.

**Comment by krig** [comment]
Alright, removed the indentation and whitespace changes.

File: heartbeat/ocf.py [hunk 1]
```diff
@@ -135,3 +135,256 @@ def emit(self, record):
...
+		action = os.environ.get("__OCF_ACTION")
+	if action is None:
+		ocf_exit_reason("No action argument set")
+		sys.exit(OCF_ERR_UNIMPLEMENTED)
+	if action in ('meta-data', 'usage', 'methods'):
+		sys.stdout.write(str(metadata) + "\n")
+		sys.exit(OCF_SUCCESS)
+
+	check_required_params()
+	for handler, func in handlers.items():
```
Comment (oalbrigt) [inline heartbeat/ocf.py]: I think it'd be smarter to do "if action in handlers:" instead of the for-loop.

Comment (krig) [reply heartbeat/ocf.py]: yeah, I'll change it.

**Comment by krig** [comment]
I updated the commit and also added a `OCF_ACTION` variable set to `$__OCF_ACTION` if set or else if only one command line argument is provided, to that command line argument.

I'm going to dig into how `ocft` works to see if there is anything needed to make python agents compatible with it as well.

**Comment by oalbrigt** [comment]
Sounds good.

**Comment by krig** [comment]
Everything seems ocft-compatible and OCF-compliant as far as I can tell, yeah.

**Comment by krig** [comment]
Yeah, I considered that. I’ll change that.

**Comment by oalbrigt** [comment]
Perfect. Thanks.

**Comment by krig** [comment]
Thanks!

File: doc/writing-python-agents.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
+# Resource Agent guide for Python
```
Comment (oalbrigt) [inline doc/writing-python-agents.md]: @krig I think we should move this under dev-guides.

Comment (krig) [reply doc/writing-python-agents.md]: Sure, I'll make a PR.

**Comment by krig** [comment]
Done: #1242

**Pull request merged**